### DISCUSSION TITLE
Microphone Selection on Android

### DISCUSSION THREAD

**Discussion by Simon818** [BearWare/TeamTalk5#2263] [Ideas]
### Description

On Android when a headset with a microphone is plugged in, TeamTalk seems to use that microphone by default. However, most Android phones have really good internal microphones, which is more than can be said for most cheap earbuds. Also, it's common for cheap earbuds to leak the headphone output back into the microphone, which can result in some feedback but could also allow others in the channel to hear the person's screen reader if they're using one.

The easiest way to bypass this problem is to use a headphone splitter, which typically doesn't have the microphone ring on it. This effectively disconnects the microphone from the phone. Strangely, the next-easiest is to buy Bluetooth earbuds, since TeamTalk allows you to disable the Bluetooth microphone. There used to be an app called [Lesser AudioSwitch](https://play.google.com/store/apps/details?id=com.nordskog.LesserAudioSwitch&hl=en_CA&gl=US) but it doesn't work with recent versions of Android.

If it's relatively trivial to list and select an audio device on Android, I'd love to see that as an option. If I missed something and I can do that already, feel free to close this.

### Application

- [ ] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkServer

**Comment by bear101** [comment]
I believe there's already an open issue regarding this.
